### ISSUE TITLE
 Support autocompletion of HTML tags and attributes

### ISSUE THREAD

**Issue by fernandreu** [fernandreu/wpf-custom-ui-editor#15]
The tool already has information about the XML schema, so this could be used for autocompletion purposes for both HTML tags and attributes.

**Comment by nbelyh** [comment]
It would be great to have autocomplete! Although think that scintilla is actually awful choice to develop autocomplete, it's kind of a nightmare 😄 
I would dump this dinosaur for good and opt for something more modern, like [monaco-editor](https://microsoft.github.io/monaco-editor/) in embedded browser (thinking [WebView2](https://docs.microsoft.com/en-us/microsoft-edge/webview2/)) that has autocomplete built-in.

**Comment by fernandreu** [comment]
Yeah, scintilla is a good replacement of the original custom UI editor but still miles behind of the level of language support you'd get in VSCode (which is also one of my favourite code editors). I'd also like to find an alternative that is at least multi-platform so that I can experiment with having a Linux / Mac version of the tool.

The problem is, switching to the Monaco editor will most likely mean rewriting everything from scratch in JavaScript / TypeScript. I'm not particularly keen on that, mostly due to the amount of work it would mean. It would probably deserve a completely separate repo too, perhaps coexisting with this one.

Perhaps something with a bit more potential would be creating an extension for VSCode itself. There might be some challenges but, if done well, you could get full XML support almost out of the box (or perhaps with additional extensions). It's definitely something I want to try at some point.

**Comment by nbelyh** [comment]
Yes, sounds interesting actually, maybe as VS code extension that would be more reasonable. There is a new XML schema-based completion extension appeared finally recently not sure if it works though: https://github.com/rogalmic/vscode-xml-complete

I did not mean to rewrite in typescript/javascript, just to embed the webbrowser (with editor inside) as a replacement for the scintilla or rich text control. Regarding embedding, WebView2 (windows), also CEF (chromium embedded framework, but this one is heavy). Also the new one emerged recently SciterJS (https://github.com/c-smile/sciter-js-sdk)

**Comment by fernandreu** [comment]
I've spent a bit of time thinking about the different options, and I think I'll try embedding the Monaco editor first.

Doing the C# / JS interop would be difficult if I had to do all that by myself from scratch, but I am hoping I can use this project for that: https://github.com/unoplatform/uno.monaco-editor-uwp

I could either replicate something similar in WPF, or just embed that UWP control as it is, which in theory it should be possible. I'll try the latter first.
